lazyflow/operators/valueProviders.py

```diff
@@ -496,3 +496,29 @@ def execute(self, slot, subindex, roi, result):
     def propagateDirty(self, slot, subindex, roi):
         if slot == self.Input:
             self.Output.setDirty(roi)
+
+
+class OpZeroSource(Operator):
+    """Operator that serves zeros on it's Output
+
+    This operator will never set anything dirty as it cannot be changed.
+    """
+
+    Output = OutputSlot()
+
+    def __init__(self, dtype, shape, *, graph=None, parent=None, **kwargs):
+        super().__init__(graph=graph, parent=parent)
+
+        self._meta = {"dtype": dtype, "shape": shape}
+        self._meta.update(kwargs)
+
+    def setupOutputs(self):
```

Comment:
Can we move this logic into the constructor? We can leave an empty `setupOutputs` method if lazyflow demands it, but since this operator has no Inputs it'll never be called anyway, and then we can move closer to a world of meaningful constructors in ilastik. As a bonus, we don't even have to keep the private `meta` dict in the operator, as `dtype` and `shape` can both be used straight from the constructor params into the `Output.meta` dict

Reply:
yes, that makes sense for this operator. But just to clarify - `setupOutput` method _is_ in fact called (Otherwise the tests would also fail).